### PR TITLE
Update the documentation of the BLE module

### PR DESCRIPTION
**Pull request by LaszloLango**
Related issue #873

TODOs:
- [x] Characteristic
- [x] Descriptor
- [x] HCI socket
- [x] Primary service

IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by daeyeon** [comment]
Thanks for listing todos.

File: docs/api/IoT.js-API-BLE.md [hunk 1]
```diff
@@ -9,64 +9,42 @@ The following shows ble module APIs available for each platform.
...
-### `ble.startAdvertising(name, serviceUuids[, callback])`
-* `name <String>` - maximum 26 bytes
-* `serviceUuids <Array>`
+### ble.startAdvertising(name, serviceUuids[, callback])
+* `name` {string} - maximum 26 bytes
+* `serviceUuids` {Array}
   * 1 128-bit service UUID
   * 1 128-bit service UUID + 2 16-bit service UUIDs
   * 7 16-bit service UUID
-* `callback <Function(error)>`
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Passed as list of bytes?

File: docs/api/IoT.js-API-BLE.md [hunk 2]
```diff
@@ -9,64 +9,42 @@ The following shows ble module APIs available for each platform.
...
+* `callback` {Function}
+  * error {Error}
 
 Starts advertising.
 
-ble.state must be poweredOn before advertising is started. 
-ble.on('stateChange', callback(state)); can be used register for state change events.
+`ble.state` must be poweredOn before advertising is started.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: must be in poweredOn state

File: docs/api/IoT.js-API-BLE.md [hunk 3]
```diff
@@ -9,64 +9,42 @@ The following shows ble module APIs available for each platform.
...
+`ble.on('stateChange', callback(state));` can be used to register for state change events.
 
 
-### `ble.stopAdvertising(callback)`
-* `callback <Function(error)>`
+### ble.stopAdvertising(callback)
+* `callback` {Function}
+  * error {Error}
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: I also added explanations for callbacks in the GPIO module. Mostly the same text with little modifications, you can copy them here.

File: docs/api/IoT.js-API-BLE.md [hunk 4]
```diff
@@ -9,64 +9,42 @@ The following shows ble module APIs available for each platform.
...
-
-### `ble.setServices(services[, callback])`
-* `services <Array>`
-* `callback <Function(error)>`
+### ble.setServices(services[, callback])
+* `services` {Array}
+* `callback` {Function}
+  * error {Error}
 
 Sets the primary services available on the peripheral.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: What is a service? String?

File: docs/api/IoT.js-API-BLE.md [hunk 5]
```diff
@@ -104,4 +84,97 @@ ble.on('advertisingStart', function(error) {
...
+    * `offset` {number} (0x0000 - 0xffff)
+    * `withoutResponse` {boolean}
+    * `callback` {Function}
+      * `result` {Characteristic.RESULT_*}
+  * `onSubscribe` {Function} optional notify/indicate subscribe handler
+    * `maxValueSize` {number} maximum data size
+    * `updateValueCallback` {Function}
+  * `onUnsubscribe` {Function} optional notify/indicate unsubscribe handler
+  * `onNotify` {Function} optional notify sent handler
+  * `onIndicate` {Function} optional indicate confirmation received handler
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Would be good some explanation about this.

File: docs/api/IoT.js-API-BLE.md (hunk 1 above)
Comment (LaszloLango) [inline docs/api/IoT.js-API-BLE.md]: This line was removed. Did you comment on the right line?

File: docs/api/IoT.js-API-BLE.md (hunk 5 above)
Comment (LaszloLango) [inline docs/api/IoT.js-API-BLE.md]: What do you mean?

Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: No description after the arguments.

File: docs/api/IoT.js-API-BLE.md (hunk 1 above)
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Sorry, refers to the above line. How do you pass a 128 bit value as an array? Would be good to add description after the arguments.

File: docs/api/IoT.js-API-BLE.md (hunk 5 above)
Comment (LaszloLango) [inline docs/api/IoT.js-API-BLE.md]: I'm still confused. There is a description. `optional indicate confirmation received handler`

**Comment by LaszloLango** [comment]
@daeyeon @zherczeg I've updated the PR please check.

File: docs/api/IoT.js-API-BLE.md [hunk 6]
```diff
@@ -9,64 +9,50 @@ The following shows ble module APIs available for each platform.
...
-
-## Module Functions
-
-
-### `ble.startAdvertising(name, serviceUuids[, callback])`
-* `name <String>` - maximum 26 bytes
-* `serviceUuids <Array>`
+### ble.startAdvertising(name, serviceUuids[, callback])
+* `name` {string} - maximum 26 bytes
+* `serviceUuids` {[String]}
```
Comment (daeyeon) [inline docs/api/IoT.js-API-BLE.md]: Looks unclear for me. I put my suggestion, `{Array.<string>}` in #873. Let me know your thought.

File: docs/api/IoT.js-API-BLE.md [hunk 7]
```diff
@@ -87,10 +73,11 @@ ble.on('stateChange', function(state){
 });
 ```
 
-### `'advertisingStart'`
-* `callback <Function(error)>`
+### Event: 'advertisingStart'
+* `callback` {Function}
+  * error {Error}
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Events should be listed first.

File: docs/api/IoT.js-API-BLE.md [hunk 8]
```diff
@@ -9,64 +9,50 @@ The following shows ble module APIs available for each platform.
...
 
-## Events
-
-
-### `'stateChange'`
-* `callback <Function(state)>`
-  * `state <String>` - "unknown" | "resetting" | "unsupported" | "unauthorized" | "poweredOff" | "poweredOn"
+### Event: 'stateChange'
+* `callback` {Function}
+  * `state` {String} - "unknown" | "resetting" | "unsupported" | "unauthorized" | "poweredOff" | "poweredOn"
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: No dash, full sentence.

File: docs/api/IoT.js-API-BLE.md [hunk 9]
```diff
@@ -9,64 +9,50 @@ The following shows ble module APIs available for each platform.
...
 
 Stops advertising.
 
 
-### `ble.setServices(services[, callback])`
-* `services <Array>`
-* `callback <Function(error)>`
+### ble.setServices(services[, callback])
+* `services` {[PrimaryService]}
+* `callback` {Function} error handler
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Full sentence. Error handler.

File: docs/api/IoT.js-API-BLE.md [hunk 10]
```diff
@@ -9,64 +9,50 @@ The following shows ble module APIs available for each platform.
...
   * 7 16-bit service UUID
-* `callback <Function(error)>`
+* `callback` {Function} error handler
+  * `error` {Error}
 
 Starts advertising.
 
-ble.state must be poweredOn before advertising is started. 
-ble.on('stateChange', callback(state)); can be used register for state change events.
+`ble.state` must be in poweredOn before advertising is started.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: poweredOn state before

File: docs/api/IoT.js-API-BLE.md [hunk 11]
```diff
@@ -9,64 +9,50 @@ The following shows ble module APIs available for each platform.
...
+  * `error` {Error}
 
 Stops advertising.
 
 
-### `ble.setServices(services[, callback])`
-* `services <Array>`
-* `callback <Function(error)>`
+### ble.setServices(services[, callback])
+* `services` {[PrimaryService]}
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Array?

File: docs/api/IoT.js-API-BLE.md [hunk 12]
```diff
@@ -104,4 +91,101 @@ ble.on('advertisingStart', function(error) {
...
+```
+
+
+## Class: Descriptor
+
+Descriptors are defined attributes that describe a characteristic value.
+
+### new Descriptor(options)
+* `options` {Object}
+  * `uuid:` {string} A Universally Unique ID (UUID) is a 16 or 128-bit value used to identify the type of every attribute.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Is this a hex string? Would worth a description in the main text part (still missing).

Comment (LaszloLango) [inline docs/api/IoT.js-API-BLE.md]: IMHO it is obvious if you check the example, but I can add it.

File: docs/api/IoT.js-API-BLE.md [hunk 13]
```diff
@@ -9,99 +9,183 @@ The following shows ble module APIs available for each platform.
...
+        // service data
+    ]);
+  }
+});
+```
 
-## Module Functions
+### Event: 'stateChange'
+* `callback` {Function}
+  * `state` {String} can be 'unknown', 'resetting', 'unsupported', 'unauthorized', 'poweredOff' or 'poweredOn'.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-BLE.md]: Can be

**Comment by LaszloLango** [comment]
Thanks for the reviews. I've updated the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**